addons/website/static/src/scss/compatibility/bs3_for_12_0.scss

```diff
@@ -0,0 +1,300 @@
+    @include text-emphasis-variant(".text-#{$old}", gray($new));
+}
+
+// Restore media ?
+
+// Restore progress bars
+@each $color, $value in $theme-colors {
+    @include bg-variant(".progress-bar-#{$color}", $value);
+}
+
+// Adapt radio ?
+
+// Adapt labels
+.label {
+    @extend .badge;
+}
+@each $color, $value in $theme-colors {
+    .label-#{$color} {
+        @include badge-variant($value);
+    }
+}
+.label-default {
+    @include badge-variant(theme-color('secondary'));
+}
+
+// Adapt center-block
+.center-block {
+    display: block;
+    margin: auto;
+}
+
+// Adapt pull-* classes
+.pull-left {
+    float: left;
+}
+.pull-right {
+    float: right;
+}
+
+// Adapt pagination
+.pagination > li {
+    @extend .page-item;
+    > a {
+        @extend .page-link;
+    }
+}
+
+// Adapt carousel
+.carousel .item {
+    @extend .carousel-item;
+}
+
+// Adapt checkboxes ?
+
+// Adapt tables
+.table-condensed {
+    @extend .table-sm;
+}
+
+// Adapt forms
+.control-label {
+    @extend .col-form-label;
+}
+.help-block {
+    @extend .form-text;
+}
+.has-error .form-control {
+    @extend .is-invalid;
+}
+.has-success .form-control {
+    @extend .is-valid;
+}
+.form-horizontal .form-group {
+    @extend .row;
+}
+
+// Adapt list-inline
+.list-inline > li {
+    @extend .list-inline-item;
+}
+
+// Adapt list-group
+.panel .list-group {
+    @extend .list-group-flush;
+}
+
+// Adapt image utilies
+.img-rounded {
+    @extend .rounded;
+}
+.img-circle {
+    @extend .rounded-circle;
+}
+
+// Adapt input group
+.input-group {
+    .input-group-btn:first-child {
+        @extend .input-group-prepend;
+    }
+    .form-control ~ .input-group-btn {
+        @extend .input-group-append;
+    }
+    .input-group-addon {
+        @extend .input-group-append;
+        @extend .input-group-text;
+    }
+}
+
+// Adapt panels
+.panel {
+    @extend .card;
+}
+@each $color, $value in $theme-colors {
+    @include bg-variant(".panel-#{$color}", $value);
+}
+@include bg-variant(".panel-default", $white);
+.panel-heading {
+    @extend .card-header;
+}
+.panel-body {
+    @extend .card-body;
+}
+.panel-footer {
+    @extend .card-footer;
+}
+.well {
+    @extend .card;
+    @extend .card-body;
+}
+
+// Adapt grid (push-pull, xs ?)
+@each $breakpoint in map-keys($grid-breakpoints) {
+    $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
+    $infix: if($infix != "", $infix, "-xs");
+
+    @include media-breakpoint-up($breakpoint, $grid-breakpoints) {
+        // `$grid-columns - 1` because offsetting by the width of an entire row isn't possible
+        @for $i from 0 through ($grid-columns - 1) {
+            .col#{$infix}-offset-#{$i} {
+                @include make-col-offset($i, $grid-columns);
+            }
+        }
+    }
+}
+
+// Adapt breadcrumb
+.breadcrumb > li {
+    @extend .breadcrumb-item;
+}
+
+// Adapt nav
+.nav > li {
+    @extend .nav-item;
+    > a {
+        @extend .nav-link;
+    }
+}
```

Comment:
I suggest this in order to fix missing support for `.navbar-right`:
```scss
.navbar-right {
    margin-left: auto !important;
}
```

Reply:
@odony I did not add any navbar support because:
- It is not possible to correctly mimic the design from the old BS3 structure (even your `navbar-right` rule would break some valid BS3 structure)
- `navbar` components was probably not used in non-updatable contents

But your rule is probably right in common structures (probably with a media query though), so if it helps... :)